### PR TITLE
Create S3 bucket for storing CloudFormation templates

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
For now, the only use case is for storing the templates from the [aws-genomics-workflows-library](https://github.com/Sage-Bionetworks-Workflows/aws-genomics-workflows-library) repository. 

@zaro0508: Is there a reason why you deploy templates to an S3 bucket instead of downloading from GitHub directly in the sceptre pre-hooks? If you think the S3 deployment is essential, then your review of this PR would be appreciated.

### CODE REVIEW

**Comment by zaro0508** [comment]
> @zaro0508: Is there a reason why you deploy templates to an S3 bucket instead of downloading from GitHub directly in the sceptre pre-hooks? If you think the S3 deployment is essential, then your review of this PR would be appreciated.

Nope, you don't need to download from S3.  You can download directly from raw github file references when using sceptre.  Cases where you may want to put templates in S3 is when you use a cloudformation type that only supports S3 references.  For example lambda code, includes of template snippets, and few other things that i can't recall ATM.  

The question here is whether you want your own cloudformation repo or do you just want to use the already existing Sage CFN repo? https://bootstrap-awss3cloudformationbucket-19qromfd235z9.s3.amazonaws.com/index.html

**Review by tthyer — CHANGES_REQUESTED** [review]
@BrunoGrandePhD I think I prefer just using a raw github url

**Comment by BrunoGrandePhD** [comment]
I've updated this PR to use the raw GitHub URL pinned to a version of the repository. This PR will also serve to test the `aws-deploy` CI workflow that was just merged. At the moment, the CI is failing because sceptre returns exit code 1 as long as the `prod` stack group is empty. 

/cc @tthyer @zaro0508

**Comment by BrunoGrandePhD** [comment]
> The question here is whether you want your own cloudformation repo or do you just want to use the already existing Sage CFN repo? https://bootstrap-awss3cloudformationbucket-19qromfd235z9.s3.amazonaws.com/index.html

@zaro0508: I think our own bucket would be ideal because I want to leverage the existing [Travis CI deployment workflow](https://github.com/Sage-Bionetworks-Workflows/aws-genomics-workflows-library/blob/master/.travis.yml) in the `aws-genomics-workflows-library` repository and I would like to keep that separate from the existing CFN S3 bucket.

File: config/config.yaml
Comment (zaro0508) [inline config/config.yaml:4]: I suggest the root be `https://raw.githubusercontent.com/Sage-Bionetworks` because there are things outside of aws-infra you might be interested, like lambdas published from Sage-Bionetworks-IT org.  Also you may want to reference different versions of  the CFN templates in each one of your sceptre templates.

File: config/prod/s3-cfn-templates.yaml
Comment (zaro0508) [inline config/prod/s3-cfn-templates.yaml:9]: please use `wget` instead of `curl`.  it's better because it has retries built in. here's an example, https://github.com/Sage-Bionetworks-IT/organizations-infra/blob/master/sceptre/scicomp/config/prod/rotate-credentials.yaml#L12

**Review by zaro0508 — CHANGES_REQUESTED** [review]

File: config/config.yaml
Comment (zaro0508) [reply config/config.yaml:4]: ohh wait, sorry.  ignore this comment.  i got mixed up with github reference and S3 bucket reference.  this is ok.

Comment (BrunoGrandePhD) [reply config/config.yaml:4]: Got it!

Comment (zaro0508) [reply config/config.yaml:4]: sorry, the version thing i said in the first comment if valid, so i think the root should be.. `https://raw.githubusercontent.com/Sage-Bionetworks/aws-infra`

**Comment by BrunoGrandePhD** [comment]
@zaro0508: I switched to `wget`. Thanks for the tip!

**Review by zaro0508 — APPROVED** [review]

**Comment by BrunoGrandePhD** [comment]
@tthyer: I switched to the GitHub URL. I'll still wait for your approval before merging.

**Review by tthyer — APPROVED** [review]

**Pull request merged**
